src/templates/java/JavaTemplateProvider.ts

```diff
@@ -72,6 +74,13 @@ export class JavaTemplateProvider extends ScriptTemplateProvider {
         return 'Java';
     }
 
+    private validateGradleProject(): void {
+        const buildTool: JavaBuildTool | undefined = getWorkspaceSetting(javaBuildTool, this.getProjectPath());
+        if (buildTool === JavaBuildTool.gradle) {
+            throw new Error(localize('gradleUpdateTemplate', 'Update function template is not supported for gradle project.'));
```

Comment:
I don't think `getLatestTemplates` is ever called directly by the user.  I think it gets called when we refresh the templates or try to back them up.  So I think that this error would be internal because I don't think that the user would have any action for this error.